shared/src/mobileMain/kotlin/dev/johnoreilly/confetti/SessionsViewModel.kt

```diff
@@ -128,28 +78,38 @@ open class SessionsViewModel : KMMViewModel(), KoinComponent {
 
     @NativeCoroutinesState
     val uiState: StateFlow<SessionsUiState> = responseDatas.receiveAsFlow()
-        .flatMapLatest { refreshData ->
-            val bookmarksData = refreshData.bookmarksResponse.data
-            repository.watchBookmarks(conference, bookmarksData)
-                .map { refreshData.copy(bookmarksResponse = it) }
-                .onStart {
-                    emit(refreshData)
-                }.map {
-                    uiStates(it)
-                }
-        }.combine(searchQuery) { uiState, search ->
+        .uiState()
+        .combine(searchQuery) { uiState, search ->
             filterSessions(uiState, search)
         }
         .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), SessionsUiState.Loading)
 
-
     // FIXME: can we pass that as a parameter somehow
-    fun configure(conference: String) {
+    fun configure(conference: String, uid: String?, tokenProvider: TokenProvider?) {
+        val hasChanged = this.conference != conference || this.uid != uid
         this.conference = conference
-        maybeStart()
+        this.uid = uid
+        this.tokenProvider = tokenProvider
+
+        refresh(showLoading = hasChanged, forceRefresh = false)
+    }
+
+    private var job: Job? = null
+
+    /**
+     * @param showLoading whether to show a loading state (if user or conference changed)
+     * @param forceRefresh whether to force a network refresh (pull-to-refresh)
+     */
+    fun refresh(showLoading: Boolean, forceRefresh: Boolean) {
+        job?.cancel()
+        job = viewModelScope.coroutineScope.launch {
+            responseData(showLoading, forceRefresh).collect {
+                responseDatas.send(it)
+            }
+        }
     }
 
-    suspend fun refresh() = refresh(false)
+    suspend fun refresh() = refresh(false, true)
```

Comment:
This was a bit hard to follow to be honest, maybe it would be nice with a named argument IMO at least here :)